system-int-test/test/cmr/system_int_test/search/granule_shapefile_simplification_search_test.clj

```diff
@@ -0,0 +1,216 @@
+(ns cmr.system-int-test.search.granule-shapefile-simplification-search-test
+  (:require
+   [clojure.test :refer :all]
+   [clojure.java.io :as io]
+   [cheshire.core :as json]
+   [cmr.common.log :refer [debug info]]
+   [cmr.common.mime-types :as mt]
+   [cmr.common.util :as util :refer [are3]]
+   [cmr.common-app.test.side-api :as side]
+   [cmr.search.services.parameters.converters.shapefile :as shapefile]
+   [cmr.search.middleware.shapefile :as shapefile-middleware]
+   [cmr.spatial.line-string :as l]
+   [cmr.spatial.mbr :as m]
+   [cmr.spatial.point :as p]
+   [cmr.spatial.polygon :as poly]
+   [cmr.system-int-test.data2.core :as d]
+   [cmr.system-int-test.data2.granule :as dg]
+   [cmr.system-int-test.data2.umm-spec-collection :as data-umm-c]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.umm.umm-spatial :as umm-s]))
+
+(use-fixtures :each (ingest/reset-fixture {"provguid1" "PROV1"}))
+
+(defn- polygon
+  "Creates a single ring polygon with the given ordinates. Points must be in counter clockwise order.
+  The polygon will be closed automatically."
+  [& ords]
+  (poly/polygon [(apply umm-s/ords->ring ords)]))
+
+(def formats
+  "Shapfile formats to be tested"
+  ;; ESRI is handled separately
+  {"GeoJSON" {:extension "geojson" :mime-type mt/geojson}
+   "KML" {:extension "kml" :mime-type mt/kml}})
+
+(deftest granule-shapefile-simlpification-failure-cases
+  (side/eval-form `(shapefile/set-enable-shapefile-parameter-flag! true))
+  (let [saved-shapefile-max-size (shapefile-middleware/max-shapefile-size)
+        _ (side/eval-form `(shapefile-middleware/set-max-shapefile-size! 2500000))]
```

Comment:
Again this looks like a different max size from the documentation and from the collection test.

Reply:
This one is even bigger than the one used for collections (or below for the other granule tests) because it uses a really complicated test file that is > 2MB. Again, I'm not testing the max limit code here, so this gets around it so I can test the max features code with a realistic test file.